imxrt-ral/imxrtral.py

```diff
@@ -39,8 +39,14 @@
 
 #![no_std]
 
+#[cfg(feature="rt")]
+extern crate imxrt_rt;
+
 mod register;
 
+#[cfg(feature="rt")]
+pub use imxrt_rt::interrupt;
+
```

Comment:
I ran into an issue with the interrupt macro not working correctly unless doing it this way, I'll look again. It is a feature flag include so it is still an optional dependency. There's some minor differences in that the imxrt-rt linker script here that match what mcuxpresso and zephyr do. I'll see if I can remove the dependency and not break things again when I get closer to merging.

Reply:
Although it's behind a feature flag, I need to enable that feature flag in the `teensy4-bsp` [to link in the interrupt table](https://github.com/imxrt-rs/imxrt-rs/blob/87e5408f1d60f1561abeec120c8fc2f6fc949072/imxrt-ral/src/imxrt106/imxrt1062/interrupts.rs#L170). The interrupt table is still supplied by the RAL. But, it doesn't need to be paired with runtime until link time.

Enabling the `rt` flag while there's also a runtime implementation will result in duplicate symbols of `_reset` and the other ARM exceptions.

Reply:
Understandable, that same issue bit me before with device.x being hidden behind a feature flag.

Do we really even need the rt feature flag in ral?

Reply:
I _think_ we still want the feature flag. Consider a Rust developer who only wants to build a library against the RAL or the HAL (something like device-specific peripheral adapters, or an alternative HAL that has one peripheral per crate). The library is used in a larger work, and the larger work enables the `rt` flag somewhere in its dependency tree. That library should not compile the interrupt table. By eliding the `rt` flag, the library designer can depend on the RAL / HAL, building their library without compiling unnecessary components and possibly duplicating the unmangled `__INTERRUPTS` symbol.

That's at least how I'm justifying it to myself; I'm not entirely sure if that's why there's a `rt` flag. I've been thinking about asking the `cortex-m-rt` team, and the broader embedded Rust group, about these design decisions and trade-offs. Thanks for riffing on these thoughts with me!

Reply:
Or, we gut the interrupt table from the RAL, and push it into the runtime crate. Our runtime crate is device-specific anyway. If we went that way, then I think we could remove the `rt` feature flag 😛 

I'm guessing that the embedded Rust group went with the current approach, since the `cortex-m-rt` crate is intended to be maximally generic. It can't know of processor-specific interrupt tables. Rather than producing both a PAC crate and an `imxrt-vector-table` crate, `svd2rust` just produces the PAC crate, and conditionally enables what would be an `imxrt-vector-table` crate in the PAC.

Reply:
I'm fine keeping the rt feature flag for now. I really want to get a 0.1.0 of this bsp out, need to fix an interrupt bug and add the debug uart to feel good about it though. Those are the last steps I feel like for a 0.1 for me.